src/psyclone/dynamo0p3.py

```diff
@@ -3934,9 +3935,34 @@ def create_halo_exchanges(self):
                                 "all dependent nodes to be halo exchanges")
                 prev_node = prev_arg_list[0].call
                 if not isinstance(prev_node, DynHaloExchange):
-                    # previous dependence is not a halo exchange so
-                    # create a new halo exchange
-                    self._add_halo_exchange(halo_field)
+                    # Previous dependence is not a halo exchange.
+                    #
+                    # There is still one case where we do not want to
+                    # add in a halo exchange. This is when the field
+                    # is continuous but only its annexed dofs are read
+                    # and the previous write is to the level 1 halo
+                    # (or more).
+                    if self._upper_bound_name == "ncells":
+                        # Only this fields annexed dofs are read. We also
+                        # now know that this is a continuous field.
```

Comment:
Please explain (in this comment) how we know that this is a continuous field.

Reply:
oh, and s/field/field's/

Reply:
This code has now been removed.